docs/main/05-reference/06-workflows.md

```diff
@@ -0,0 +1,159 @@
+      env: []
+      envFrom: []
+      volumeMounts: []
+```
+
+Refer to the [Kubernetes Pod Spec
+documentation](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
+for more information on the fields above.
+
+To access secrets in the pipeline, refer to the [Kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+:::note
+
+Not all fields from the Pod spec are supported. We will add support for more
+fields in the future.
+
+:::
+
+## Volumes {#volumes}
+
+Kratix will run each container in the `spec.containers` list in order,
+providing a set of common volumes, as defined below.
+
+### Input
+
+Kratix provides a input directory to the container at `/kratix/input`. This
+directory is populated with different files depending on the type of Workflow.
+
+In Promise Workflows, this directory is empty.
+
+In Resource Workflows, all container will have access to an `object.yaml`
+within the `/kratix/input` directory. The `object.yaml` contains the Resource
+definition submitted to the platform.
+
+### Output
+
+At the end of the workflow, all files present in the output volume, mounted at
+`/kratix/output`, will be written to the State Store. All containers in the
+pipeline can write to this volume, and any container can add, update, or remove
+documents from this directory.
+
+In Promise Workflows, if the Promise specifies additional `.spec.dependencies`,
+these will be automatically added to the directory at the beginning of the
+pipeline at `/kratix/output/static/dependencies.yaml`.
+
+:::note
+
+Files written to `/kratix/output` in `delete` pipelines will be ignored.
+
+:::
+
+### Metadata
+
+All containers in the `configure` Pipeline have access to
+this directory, mounted at `/kratix/metadata`.
+
+Pipeline containers can control aspects of how Kratix behaves by creating
+special files in this directory:
+
+- `destination-selectors.yaml` can be added to any Promise to
+  further refine where the resources in `/kratix/output` will be
+  [scheduled](./multicluster-management).
+- `status.yaml` allows the Pipeline to communicate information about the
+  Resource back to the requester. See [status documentation
+  for more information](./resources/status).
+
+Kratix scans for these files and ignores all other files in the `/kratix/metadata`
+directory. If you need to pass additional information to the next container in
+the pipeline, you can safely write to the `/kratix/metadata` directory.
+
+## Environment Variables {#environment-variables}
+
+Kratix will set the following environment variables for all containers in the
+workflow:
+
+
+| Variable                  | Description     |
+| ------------------------- | --------------- |
+| `KRATIX_WORKFLOW_ACTION`  | The action that triggered the workflow. Either `configure` or `delete`. |
+| `KRATIX_WORKFLOW_TYPE`    | The type of workflow. Either `resource` or `promise`. |
+| `KRATIX_PROMISE_NAME`     | The name of the Promise. |
```

Comment:
I think its `KRATIX_PROMISE`

Reply:
maybe it should be `KRATIX_PROMISE_NAME` though. FWIW I jsut added this last friday to help my research day project lol

Reply:
i started writing `KRATIX_PR` and copilot autocompleted. i confess i didn't check 🙈

Reply:
i think kratix_promise_name is better btw

Reply:
go for it, i can update my research day work